tests/ops/qubit/test_attributes.py

```diff
@@ -436,3 +435,149 @@ def test_qaoa_embedding(self, features, weights, num_wires, batch_size):
             features, weights, wires=list(range(num_wires)), local_field=qml.RY
         )
         op.decomposition()
+
+
+class TestControlledSupportsBroadcasting:
```

Comment:
Nice!! 🚀 

I realised there are some tests from `TestSupportsBroadcasting` that are not being tested here. Is it because their controlled version is not supported?

Reply:
Also, should we move this to `test_controlled_op.py`?

Reply:
I didn't think we would want to do a `Controlled` version of embeddings or state vectors, so I didn't test them - but that might be wrong. I just tried them and they work. Should they be included in the tests?

I think moving them would make sense.

Reply:
Adding a test doesn't hurt, thus I would include them.

Let's also move the tests then!

Reply:
I'm a little confused about why the `Embedding` and `StateVector` tests all include something along the lines of 

```
qml.QubitStateVector.compute_decomposition(state, list(range(num_wires)))
op.decomposition()
```

I'm not sure what that is testing, or whether I should include something similar in the broadcasting tests for the controlled version. Thoughts?

Reply:
I believe it is testing that `compute_decomposition` doesn't raise any errors when using batching

Reply:
I would also include it in the broadcasting tests of the controlled version